### ISSUE TITLE
Off by one if Yum plugins are in use

### ISSUE THREAD

**Issue by bjvrielink** [gabe-sky/puppet-yum_updates#3]
The number of updates available should be lowered by 1 if any yum plugins are enabled:

```
[root@localhost ~]# yum check-update --quiet 

epel-release.noarch                                                                                7-12                                                                                Xxxx_el7_repos_epel_x86_64
Loaded plugins: fastestmirror, product-id, subscription-manager
[root@localhost ~]# facter -p yum_updates
{
  count => 2,
  available => true
}
[root@localhost ~]# yum update -y --quiet
Loaded plugins: fastestmirror, product-id, subscription-manager
Loaded plugins: fastestmirror, product-id, subscription-manager
Loaded plugins: fastestmirror, product-id, subscription-manager
Loaded plugins: fastestmirror, product-id, subscription-manager
Uploading Tracer Profile

Updated:
  epel-release.noarch 0:7-12                                                                                                                                                                                       

Complete!
Uploading Enabled Repositories Report
Loaded plugins: fastestmirror, product-id, subscription-manager
[root@localhost ~]# facter -p yum_updates
{
  count => 1,
  available => true
}
[root@localhost ~]# yum update -y --quiet
Loaded plugins: fastestmirror, product-id, subscription-manager
```

**Comment by gabe-sky** [comment]
Odd, `--quiet` suppresses that line for me.  I've re-done my logic significantly to scan through the output and look for lines that appear to be a package name, version, and source.  Do you think what I've got in this branch will work:

https://github.com/gabe-sky/puppet-yum_updates/blob/rework_counting/lib/facter/yum_updates.rb

Do you mind trying it on your troublesome system?  You should be able to just copy and paste that into a test system's `/opt/puppetlabs/puppet/cache/lib/facter/yum_updates.rb` and then quickly run `facter -p yum_updates` to see what it does.  (The next puppet agent run will revert it back to normal.)

**Comment by bjvrielink** [comment]
This works for me. I just had 6 updates available and after a yum update I have 0 available. Thank you for looking into this.

**Comment by gabe-sky** [comment]
Cool, thanks for testing!  I've released version 1.5.0 with the new counting method.
